docs/ref/optimizers/ceres_tutorial.md

```diff
@@ -0,0 +1,205 @@
+# Ceres Tutorial
+As with all nonlinear-optimizers they can be difficult to master, in the following we give examples on how to effectively implement custom cost functions in Ceres-Solver. In particular we will cover several different topics:
+
+- How to define a cost function?
+- How to define the optimization problem?
+- How do I define a numerical cost function?
+- How do I define an analytical cost function?
+- How do I load the optimization data?
+
+
+## How to define a cost function?
+Ceres an solve bounds constrained robustified non-linear least squares problems of the form:
+
+\begin{equation}
+  \min_{x} \dfrac{1}{2} \sum_{i} \rho_{i} (|| f_{i} (x_{i}, \dots, x_{i_{k}}) ||^{2})
+\end{equation}
+
+The expression $\rho_{i} (|| f_{i} (x_{i}, \dots, x_{i_{k}}) ||^{2})$ is known in ceres as a **`ResidualBlock`**, where $f_{i}(\cdot)$ is a cost function that depends on the parameter blocks $[x_{i}, \dots, x_{i_{k}}]$.
+
```

Comment:
Yeah, it became even more muddled below.  They refer to the residual function as the cost function, which it is, once you take the norm and square it.  Since that is the assumed form for all Ceres costs,  there is no cost function definition. They also do mention robust costs, which would imply they can change to a different cost, not just pure least squares. If you can make the distinction between residual and cost more clear in some places, that would be great.